core/remotes/docker/referrers.go

```diff
@@ -0,0 +1,99 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package docker
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/containerd/errdefs"
+	"github.com/containerd/log"
+	digest "github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+func (r dockerFetcher) FetchReferrers(ctx context.Context, dgst digest.Digest, artifactTypes ...string) ([]ocispec.Descriptor, error) {
+	rc, size, err := r.openReferrers(ctx, dgst, artifactTypes...)
+	if err != nil {
+		return nil, err
+	}
+	defer rc.Close()
+
+	var index ocispec.Index
+	if err := json.NewDecoder(io.LimitReader(rc, size)).Decode(&index); err != nil {
+		return nil, fmt.Errorf("failed to decode referrers index: %w", err)
+	}
+	return index.Manifests, nil
```

Comment:
As matching the artifactType is not guaranteed, then this should reapply the filter manually before returning.